service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RemovalToolController.java

```diff
@@ -94,36 +109,30 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
         CaseDetailsMap caseDetailsMap = caseDetailsMap(caseDetails);
         CaseData caseData = getCaseData(caseDetails);
 
-        if (caseData.getRemovableType() == APPLICATION) {
-            UUID removedApplicationId = getDynamicListSelectedValue(caseData.getRemovableApplicationList(), mapper);
+        if (caseData.getRemovalToolData().getRemovableType() == ADDITIONAL_APPLICATION) {
+            UUID removedApplicationId = getDynamicListSelectedValue(
+                caseData.getRemovalToolData().getRemovableApplicationList(), mapper);
             applicationService.removeApplicationFromCase(caseData, caseDetailsMap, removedApplicationId);
-        } else {
-            UUID removedOrderId = getDynamicListSelectedValue(caseData.getRemovableOrderList(), mapper);
+        } else if (caseData.getRemovalToolData().getRemovableType() == ORDER) {
+            UUID removedOrderId = getDynamicListSelectedValue(caseData.getRemovalToolData().getRemovableOrderList(),
+                mapper);
             RemovableOrder removableOrder = orderService.getRemovedOrderByUUID(caseData, removedOrderId);
 
             orderService.removeOrderFromCase(caseData, caseDetailsMap, removedOrderId, removableOrder);
-
+        } else {
+            if (isEmpty(caseData.getC110A().getDocument())) {
+                return respond(caseDetailsMap, List.of(APPLICATION_FORM_ALREADY_REMOVED_ERROR_MESSAGE));
+            }
+            caseDetailsMap.put("submittedForm", null);
+            caseDetailsMap.put("supplementDocument", null);
+            caseDetailsMap.put("hiddenApplicationForm", RemovedApplicationForm.builder()
+                    .submittedForm(caseData.getC110A().getDocument())
+                    .submittedSupplement(!isEmpty(caseData.getC110A().getSupplementDocument())
+                        ? caseData.getC110A().getSupplementDocument() : null)
```

Comment:
don't know if we could maybe get away with having this just as 
```suggestion
                        .submittedSupplement(caseData.getC110A().getSupplementDocument())
```?

Reply:
ah yeah true! if it's not there it'll be null anyway!